test/stdlib/StringDescribing.swift

```diff
@@ -0,0 +1,21 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+
+private class Foo {}
+class Bar {}
+
+var StringDescribingTestSuite = TestSuite("StringDescribing")
+
+StringDescribingTestSuite.test("String(describing:) shouldn't include extra stuff if the class is private") {
+  expectEqual(String(describing: Foo.self), "Foo")
+  expectEqual(String(describing: Bar.self), "Bar")
+}
+
+StringDescribingTestSuite.test("String(reflecting:) should include extra stuff if the class is private") {
+  expectEqual(String(reflecting: Bar.self), "main.Bar")
+  expectNotEqual(String(reflecting: Foo.self), "main.(private).Foo")
```

Comment:
Sorry, I meant to actually use `expectEqual` here for the current format. You can even mention in a comment that no one's relying on the details of the full demangled string, but that we want to know when it changes.

Reply:
My bad, I had not understood the first time.
But the se mangled string will be equal between different platform?

Reply:
Yes, it should be. The "private discriminator" part of the mangled name needs to be consistent on *one* platform so that rebuilding doesn't give you something different, but at the moment it's also consistent across platforms, and we have other tests that already rely on that.